lib/services/role.go

```diff
@@ -2758,7 +2758,16 @@ func (set RoleSet) checkAccess(r AccessCheckable, traits wrappers.Traits, state
 		}
 
 		// Device verification.
-		if !deviceAllowed && role.GetOptions().DeviceTrustMode == constants.DeviceTrustModeRequired {
+		deviceVerificationPassed := true
+		switch role.GetOptions().DeviceTrustMode {
+		case constants.DeviceTrustModeRequiredForHumans:
+			// Humans must use trusted devices, bots can use untrusted devices.
+			deviceVerificationPassed = deviceTrusted || state.IsBot
+		case constants.DeviceTrustModeRequired:
+			// Only trusted devices allowed for bot human and bot users.
+			deviceVerificationPassed = deviceTrusted
+		}
+		if !deviceVerificationPassed {
```

Comment:
can the default value of deviceVerificationPassed be false, and explicitly set to true if device trust is not required, so that if we later add a new device trust mode and forget to update this it won't pass by default?

Reply:
Great idea! Done in 2eeb0c144819cc42f5a958b8a38bef049800017d.